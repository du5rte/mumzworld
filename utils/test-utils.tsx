import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import { lightTheme } from '@/styles/themes';
import '@/locales/i18next-jest';

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
};

const customRender: typeof render = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
