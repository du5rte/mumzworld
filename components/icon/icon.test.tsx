import { render, screen } from '@/utils/test-utils';

import Icon from './icon';

describe('icon.tsx', () => {
  it('should render correctly', () => {
    render(<Icon name="shopping-bag" size={28} color="primary" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
