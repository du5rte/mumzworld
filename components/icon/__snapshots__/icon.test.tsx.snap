// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon.tsx should render correctly 1`] = `<Text />`;
